ios/engine/KMEI/KeymanEngine/Classes/KeymanTextField.swift

```diff
@@ -0,0 +1,355 @@
+//
+//  KeymanTextField.swift
+//  KeymanEngine
+//
+//  Created by Gabriel Wong on 2017-10-05.
+//  Copyright © 2017 SIL International. All rights reserved.
+//
+
+import AudioToolbox
+import UIKit
+
+public class KeymanTextField: UITextField, UITextFieldDelegate, KeymanWebViewDelegate {
+  // viewController should be set to main view controller to enable keyboard picker.
+  public var viewController: UIViewController?
+
+  // Sets Keyman custom font (if any) to KeymanTextField instance on keyboard change event
+  public var shouldSetCustomFontOnKeyboardChange = true
+  public var isInputClickSoundEnabled = true
+
+  private var delegateProxy: KeymanTextFieldDelegateProxy!
+  private var shouldUpdateKMText = false
+
+  // MARK: - Object Admin
+  deinit {
+    NotificationCenter.default.removeObserver(self)
+    delegate = nil
+  }
+
+  public convenience init() {
+    self.init(frame: CGRect.zero)
+  }
+
+  public override init(frame: CGRect) {
+    super.init(frame: frame)
+    performCommonInit()
+  }
+
+  public required init?(coder aDecoder: NSCoder) {
+    super.init(coder: aDecoder)
+    performCommonInit()
+  }
+
+  private func performCommonInit() {
+    delegateProxy = KeymanTextFieldDelegateProxy(self)
+    delegate = delegateProxy
+
+    if #available(iOS 9.0, *) {
+      inputAssistantItem.leadingBarButtonGroups = []
+      inputAssistantItem.trailingBarButtonGroups = []
+    }
+
+    NotificationCenter.default.addObserver(self, selector: #selector(self.textFieldTextDidChange),
```

Comment:
There was a call to [`[KMManager sharedInstance]; // Preload webview keyboard`](https://github.com/keymanapp/keyman/pull/378/files#diff-b6ef7f3c4df91bbb8a742e9a09330e20L58) here.  

I take it this is no longer needed/necessary?  We might need some testing for that to be sure.